src/nvector/serial/nvector_serial.c

```diff
@@ -205,18 +193,18 @@ N_Vector N_VNew_Serial(sunindextype length, SUNContext sunctx)
 N_Vector N_VMake_Serial(sunindextype length, sunrealtype* v_data,
                         SUNContext sunctx)
 {
+  SUNFunctionBegin(sunctx);
   N_Vector v;
 
+  SUNAssertNull(length >= 0, SUN_ERR_ARG_OUTOFRANGE);
+
   v = NULL;
   v = N_VNewEmpty_Serial(length, sunctx);
-  if (v == NULL) { return (NULL); }
+  SUNCheckLastErrNull();
 
-  if (length > 0)
-  {
-    /* Attach data */
-    NV_OWN_DATA_S(v) = SUNFALSE;
-    NV_DATA_S(v)     = v_data;
-  }
+  /* Attach data */
+  NV_OWN_DATA_S(v) = SUNFALSE;
+  NV_DATA_S(v)     = v_data;
```

Comment:
The assert above allows `length = 0`, so should these attach operations still be protected inside `if (length > 0)`?